buffer/src/main/java/io/netty/buffer/ByteBuf.java

```diff
@@ -404,20 +404,23 @@
     public abstract ByteBuf setIndex(int readerIndex, int writerIndex);
 
     /**
-     * Returns the number of readable bytes which is equal to
-     * {@code (this.writerIndex - this.readerIndex)}.
+     * Returns the number of readable bytes which is logically equivalent to
+     * {@code (this.writerIndex - this.readerIndex)}, but maybe overridden to accommodate
+     * specialized behavior (e.g. write only).
      */
     public abstract int readableBytes();
 
     /**
-     * Returns the number of writable bytes which is equal to
-     * {@code (this.capacity - this.writerIndex)}.
+     * Returns the number of writable bytes which is logically equivalent to
```

Comment:
I updated the language here to clarify expected behavior. I anticipate the risk of the change to be low, and IMHO worth it to make the APIs more representative of real usage, but recognize there is some risk. Even if someone is has a `ReadOnlyByteBuf` and is using `writeableBytes()` to guard "can I write to this buffer", they will end up getting a `ReadOnly*Exception` if this method returns non-zero today.